### DISCUSSION TITLE
Helper Script for updating Paperless-NGX including Ghostscript

### DISCUSSION THREAD

**Discussion by suxlex** [tteck/Proxmox#2180] [Help Needed]
After upgrading to paperless-ngx from version 1.17.4 to the current version 2.1.1 with the Proxmox helper script the import of documents unfortunately no longer works at all. A red window appears with the message:

[ERROR] [paperless.consumer] Error occurred while consuming document plan.pdf  "MissingDependencyError: gs"

The reason could be that Ghostsscript version 9.55 or higher is now required. 
Unfortunately, the Proxmox helper script for paperless-ngx does not automatically update Ghostsscript when updating paperless-ngx?

**Comment by tteck** [comment]
What version of Debian?

Debian 11 (gs 9.53.3)
```
apt-cache policy ghostscript
ghostscript:
  Installed: (none)
  Candidate: 9.53.3~dfsg-7+deb11u6
  Version table:
     9.53.3~dfsg-7+deb11u6 500
        500 http://deb.debian.org/debian bullseye/main amd64 Packages
     9.53.3~dfsg-7+deb11u5 500
        500 http://security.debian.org bullseye-security/main amd64 Packages
```


Debian 12 (gs 10.0.0)
```
apt-cache policy ghostscript
ghostscript:
  Installed: (none)
  Candidate: 10.0.0~dfsg-11+deb12u2
  Version table:
     10.0.0~dfsg-11+deb12u2 500
        500 http://ftp.debian.org/debian bookworm/main amd64 Packages
     10.0.0~dfsg-11+deb12u1 500
        500 http://security.debian.org/debian-security bookworm-security/main amd64 Packages
```

**Reply by suxlex** [reply]
apt-cache policy ghostscript
ghostscript:
  Installed: 9.53.3 dfsg-7+deb11u6
  Candidate: 9.53.3 dfsg-7+deb11u6
  Version table:
    9.53.3 dfsg-7+deb11u6 500
        500 http://ftp.debian.org/debian bullseye/main amd64 Packages
        100 /var/lib/dpkg/status
     9.53.3 dfsg-7+deb11u5` 500
        500 http://security.debian.org bullseye-security/main amd64 Packages

**Reply by tteck** [reply]
So, Debian 11. Maybe try updating to [Debian 12](https://github.com/tteck/Proxmox/discussions/1498) and then use the update script.

**Reply by suxlex** [reply]
Thanks, I'll try it that way.

**Reply by Lachette1** [reply]
any progress on this? Following previous suggestion, updating to Debian 12 on LXC and rerunning the update script generates this error:

>  ✓ Stopped Paperless-ngx
>  - Updating to v2.6.2  -
> [ERROR] in line 86: exit code 0: while executing command pip install -r requirements.txt &> /dev/null

**Reply by Lachette1** [reply]
I figured it out:
I did the update to Debian12 but not the same way that was listed in #1498 
Followed those instructions to deal with the externally managed python message.
Then followed the solution in #1396

Finally reran the update script.
Looks like its working well.
